parsl/tests/test_scaling/test_scale_down.py

```diff
@@ -1,80 +1,97 @@
+                    max_blocks=_max_blocks,
+                    min_blocks=_min_blocks,
                     launcher=SingleNodeLauncher(),
                 ),
             )
         ],
-        max_idletime=5,
+        max_idletime=0.5,
         strategy='htex_auto_scale',
     )
 
 
 @python_app
-def sleeper(t):
+def waiting_app(ident: int, inputs=()):
+    import pathlib
     import time
-    time.sleep(t)
+
+    # Approximate an Event by writing to files; the test logic will poll this file
+    with open(inputs[0], "a") as f:
+        f.write(f"Ready: {ident}\n")
+
+    # Similarly, use Event approximation (file check!) by polling.
+    may_finish_file = pathlib.Path(inputs[1])
+    while not may_finish_file.exists():
+        time.sleep(0.01)
 
 
 # see issue #1885 for details of failures of this test.
 # at the time of issue #1885 this test was failing frequently
 # in CI.
 @pytest.mark.local
-def test_scale_out():
-    logger.info("start")
+def test_scale_out(tmpd_cwd, try_assert):
     dfk = parsl.dfk()
 
-    logger.info("initial asserts")
-    assert len(dfk.executors['htex_local'].connected_managers()) == 0, "Expected 0 managers at start"
+    num_managers = len(dfk.executors['htex_local'].connected_managers())
+
+    assert num_managers == 0, "Expected 0 managers at start"
     assert dfk.executors['htex_local'].outstanding == 0, "Expected 0 tasks at start"
 
-    logger.info("launching tasks")
-    fus = [sleeper(i) for i in [15 for x in range(0, 10)]]
+    ntasks = 10
+    ready_path = tmpd_cwd / "workers_ready"
+    finish_path = tmpd_cwd / "workers_may_continue"
+    ready_path.touch()
+    inputs = [File(str(ready_path)), File(str(finish_path))]
 
-    logger.info("waiting for warm up")
-    time.sleep(15)
+    futs = [waiting_app(i, inputs=inputs) for i in range(ntasks)]
 
-    logger.info("asserting 5 managers")
-    assert len(dfk.executors['htex_local'].connected_managers()) == 5, "Expected 5 managers after some time"
+    while ready_path.read_text().count("\n") < _max_blocks:
+        time.sleep(0.5)
 
-    logger.info("waiting for all futures to complete")
-    [x.result() for x in fus]
+    assert len(dfk.executors['htex_local'].connected_managers()) == 5
 
-    logger.info("asserting 0 outstanding tasks after completion")
-    assert dfk.executors['htex_local'].outstanding == 0, "Expected 0 outstanding tasks after future completion"
+    finish_path.touch()  # Approximation of Event, via files
+    [x.result() for x in futs]
 
-    logger.info("waiting a while for scale down")
-    time.sleep(25)
+    assert dfk.executors['htex_local'].outstanding == 0
 
-    logger.info("asserting 2 managers remain")
-    assert len(dfk.executors['htex_local'].connected_managers()) == 2, "Expected 2 managers when no tasks, lower bound by min_blocks"
+    def assert_kernel():
+        return len(dfk.executors['htex_local'].connected_managers()) == _min_blocks
 
-    logger.info("test passed")
+    try_assert(
```

Comment:
some part of me says `try_assert` adds a lot of complexity that might be a relatively small while loop?

Reply:
Yeah, I concur on this point.  However, my thinking was that surely this won't be the only place where we need a "give it some settling time" type assertion.

As discussed out-of-band, neither of us has a particular strong global insight into all of the tests, so agreed to leave it around for now.  If it ends up being used only by this test, we can remove it at some point in the indeterminate future.